### PR TITLE
no-issue: Refactor models in server-boilerplate to export individual interfaces

### PR DESCRIPTION
**Pull request by rohan-bes**
Just decided to rework this change slightly, so that's it's easier to import specific models/records from `@tupaia/server-boilerplate`

### CODE REVIEW

File: packages/database/src/migrations/20240321045103-SetEntityMetadataNonNullable-modifies-schema.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = function (db) {
+  return db.runSql(`
+    UPDATE entity SET metadata = '{}' WHERE metadata IS NULL;
+    ALTER TABLE entity ALTER COLUMN metadata SET DEFAULT '{}';
+    ALTER TABLE entity ALTER COLUMN metadata SET NOT NULL;
+  `);
```
Comment (rohan-bes) [inline packages/database/src/migrations/20240321045103-SetEntityMetadataNonNullable-modifies-schema.js:22]: Just a migration to add a default, cos it makes life and the types nicer

File: packages/datatrak-web-server/src/types.ts [hunk 2]
```diff
@@ -2,17 +2,25 @@
...
+export interface DatatrakWebServerModelRegistry extends ModelRegistry {
+  readonly user: UserModel;
+  readonly entity: EntityModel;
+  readonly country: CountryModel;
+  readonly feedItem: FeedItemModel;
+  readonly survey: SurveyModel;
+  readonly surveyResponse: SurveyResponseModel;
+  readonly oneTimeLogin: OneTimeLoginModel;
+  readonly option: OptionModel;
+}
```
Comment (rohan-bes) [inline packages/datatrak-web-server/src/types.ts:26]: There's a bit more boilerplate, but I think I prefer this interface from a readability standpoint

File: packages/server-boilerplate/src/microService/utils/logApiRequest.ts [hunk 3]
```diff
@@ -6,21 +6,19 @@
...
+      version,
+      api: apiName,
+      method: req.method,
+      endpoint: req.path,
+      query: req.query,
+      user_id: userId,
+    });
+    req.apiRequestLogId = apiRequestLogId;
+    next();
+  };
```
Comment (rohan-bes) [inline packages/server-boilerplate/src/microService/utils/logApiRequest.ts:24]: This is just whitespace

**Review by alexd-bes — APPROVED** [review]
Thanks!

**Pull request merged**